### PR TITLE
fix: Fixed Name and Ticket crash issue

### LINKED ISSUE #1990: App crashes on selecting NAME or TICKET option in CheckIn(Attendees)
**Describe the bug**
App closes suddenly when selecting option Name or Ticket from the menu items(top right corner) in Attendees window.
**To Reproduce**
1.Create a event.
2. Open the created event.
3. Select to CHECK IN option(Attendees activity will open up)
4. From the top right options Select either Name or Ticket.
Here, the app crashes.

**Expected behavior**

**Logs**

**Screenshots**
WhatsApp Image 2019-12-06 at 7 37 36 PM

**Smartphone Info:**

|               | |
|---------------|-|
|Device         ||
|Android Version|    |

**Additional context**

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by pallav12**
Fixes #1990 

Checklist:

- I have checked for PMD and check-style issues : I don't know how to, I did install check-style plugin in android studio and it shows no error
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Changes: Issue was because of fasttItemAdapter was expecting a ComparableItemListImpl object, So created a new comparator and new ComparableItemListImpl
gif 
Attendeegif

Links for the issue : [https://github.com/mikepenz/FastAdapter/issues/852](https://github.com/mikepenz/FastAdapter/issues/852), [SortActivity.kt](https://github.com/mikepenz/FastAdapter/blob/develop/app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.kt)

### CODE REVIEW

**Comment by Saransh0905** [comment]
@pallav12  Can you tell me how did you tested whether it is sorting correctly or not?
Thanks

**Comment by pallav12** [comment]
@Saransh0905 I added Attendee in attendeeList manually in onCreateView of AtendeeFragment, for that i created a new constructor in Attendee. filtering worked perfectly on them...

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4765425&bid=15885960)

File: app/src/main/java/com/eventyay/organizer/core/attendee/list/AttendeesFragment.java [hunk 1]
```diff
@@ -79,6 +82,8 @@
 
     private static final long FIRST_PAGE = 1;
 
+    public ComparableItemListImpl<Attendee> itemListImpl;
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/attendee/list/AttendeesFragment.java]: Codacy Issue found: [Variable 'itemListImpl' must be private and have accessor methods.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4765425)

File: app/src/main/java/com/eventyay/organizer/core/attendee/list/AttendeesFragment.java [hunk 2]
```diff
@@ -88,6 +93,8 @@
 
     private static final String FILTER_SYNC = "FILTER_SYNC";
 
+    private int sorting_strategy = 0;
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/attendee/list/AttendeesFragment.java]: Codacy Issue found: [Name 'sorting_strategy' must match pattern '^[a-z][a-zA-Z0-9]*$'.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4765425)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4765425&bid=15885960)

File: app/src/main/java/com/eventyay/organizer/core/attendee/list/AttendeesFragment.java [hunk 3]
```diff
@@ -224,7 +231,20 @@ public boolean onQueryTextChange(String s) {
...
+        comparator = (o1, o2) -> {
+            if (sortingStrategy == SORTBYNAME) {
+                return o1.getFirstname().compareTo(o2.getFirstname());
+            } else if (sortingStrategy == SORTBYTICKET) {
+                return o1.getTicket().getType().compareTo(o2.getTicket().getType());
+            }
+            return 0;
+        };
+
+        ComparableItemListImpl<Attendee> itemListImpl = new ComparableItemListImpl<>(comparator);
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/attendee/list/AttendeesFragment.java]: Codacy Issue found: [Variable 'itemListImpl' should be declared final.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4765425)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4765425&bid=15885960)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4765425&bid=15885960)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4765425&bid=15885960)

**Pull request closed without merging**